Flaky test: AccessibilityServiceClient WebSocket connection failure handling

## Description

The test `AccessibilityServiceClient > getLatestHierarchy > should handle WebSocket connection failure gracefully` is flaky and occasionally fails with timing-related issues.

## Test Location

`test/features/observe/AccessibilityServiceClient.test.ts:210`

## Failure Details

**Expected behavior:** Test expects `null` when WebSocket connection fails
**Actual behavior:** Test receives cached data instead of `null`

```
AssertionError: expected { updatedAt: 1767119508829, …(2) } to be null
  at Context.<anonymous> (test/features/observe/AccessibilityServiceClient.test.ts:210:35)
```

## Observations

1. **Flakiness confirmed**: Test failed in one run but passed on immediate re-run
2. **Root cause**: Appears to be a timing/state issue in the WebSocket mock
3. **Cache contamination**: Previous test runs may be leaving cached data that affects subsequent runs

## Test Runs

### First run (failed):
```
✔ should return hierarchy data when WebSocket receives fresh data (113ms)
✔ should return cached data when not waiting for fresh data (54ms)
✔ should timeout when no data received within timeout period
✗ should handle WebSocket connection failure gracefully (1097ms)
```

### Second run (passed):
```
✔ should return hierarchy data when WebSocket receives fresh data (113ms)
✔ should return cached data when not waiting for fresh data (53ms)
✔ should timeout when no data received within timeout period
✔ should handle WebSocket connection failure gracefully (1034ms)
```

## Suggested Fixes

1. **Improve test isolation**: Ensure each test properly cleans up cached data
2. **Reset singleton state**: Clear `AccessibilityServiceClient` singleton state between tests
3. **Mock cleanup**: Verify WebSocket mocks are properly torn down
4. **Add explicit cache clearing**: Add `beforeEach` hook to clear any cached hierarchy data

## Impact

- **Severity**: Low (test failure doesn't indicate actual bug in implementation)
- **Frequency**: Intermittent (~50% failure rate observed)
- **Workaround**: Re-run tests until passing

## Additional Context

This issue was discovered while implementing #27 (scroll position handling). All scroll position tests pass reliably, and the flaky test is unrelated to that feature.

--- a/test/features/observe/AccessibilityServiceClient.test.ts
+++ b/test/features/observe/AccessibilityServiceClient.test.ts
@@ -52,6 +52,9 @@ describe("AccessibilityServiceClient", function() {
 
     accessibilityServiceClient = AccessibilityServiceClient.getInstance(testDevice, adbClient);
     AndroidAccessibilityServiceManager.getInstance(testDevice, adbClient).clearAvailabilityCache();
+
+    // Clear any cached hierarchy data to prevent cache contamination between tests (issue #72)
+    accessibilityServiceClient.invalidateCache();
   });
 
   afterEach(async function() {
@@ -200,15 +203,26 @@ describe("AccessibilityServiceClient", function() {
     });
 
     it("should handle WebSocket connection failure gracefully", async function() {
-      this.timeout(6000); // Increased to allow for 5000ms WebSocket timeout + cleanup (see issue #68)
+      // Use FakeWebSocket with instant failure and FakeTimer for fast, reliable test execution
+      // See issues #68 (timeout race condition) and #72 (cache contamination)
+      fakeTimer.setSleepDuration(1);
 
-      // Don't create a server, so connection will fail
+      const testClient = AccessibilityServiceClient.createForTesting(
+        testDevice,
+        adbClient,
+        createInstantFailureWebSocketFactory(),
+        fakeTimer
+      );
 
-      const result = await accessibilityServiceClient.getLatestHierarchy(true, 1000);
+      try {
+        const result = await testClient.getLatestHierarchy(true, 1000);
 
-      expect(result).to.not.be.null;
-      expect(result.hierarchy).to.be.null;
-      expect(result.fresh).to.be.false;
+        expect(result).to.not.be.null;
+        expect(result.hierarchy).to.be.null;
+        expect(result.fresh).to.be.false;
+      } finally {
+        await testClient.close();
+      }
     });
   });
 

